extend.php

```diff
@@ -59,6 +60,12 @@
     (new Extend\Console())
         ->command(RecalculateCriteria::class),
 
+    (new Extend\Event())
+        ->listen(Registered::class, function(Registered $event) {
+            $event->user->refreshDiscussionCount();
+            $event->user->refreshCommentCount();
+        }),
+
```

Comment:
What's the use case for these? Shouldn't discussion/comment count be 0 for newly registered users?

Reply:
This is not to break your existing drivers. They assume that replies_count and discussions_count are present on the model and return an integer, which is not the case immediately after registering (they return NULL).

Reply:
In core, the column defaults are 0: https://github.com/flarum/core/blob/c10a30bae9902ae6b869900c1e1dd89f9a6ddb95/migrations/2015_02_24_000000_create_users_table.php#L27-L28

Is the problem that the in-memory objects don't pull from the DB on save, and have `null` values as a result? If so, a better fix might be to add the default values to https://github.com/flarum/core/blob/c10a30bae9902ae6b869900c1e1dd89f9a6ddb95/src/User/User.php#L154-L154. That has its own problems (we lose a single source of truth for the default values), but a new user having no posts/discussions feels like a pretty safe invariant.

If that's the case, I would prefer to change it to `0` here to prevent the unnecessary queries on every registration, also add the default values to core, and then remove that code from this extension.

Reply:
Yes, that's the case and I agree

Reply:
Actually..! Best thing to do is to just add a little ?: 0 here
https://github.com/askvortsov1/flarum-auto-moderator/blob/master/src/Metric/PostsMade.php#L40

Reply:
Actually! Maybe metric drivers should be able to handle null values?
Or just add a little ?: 0 here? The thing is, I don't know how common it is for the metric drivers to receive a User instance that doesn't have comment_count and discussion_count populated. Should be an edge case

https://github.com/askvortsov1/flarum-auto-moderator/blob/master/src/Metric/PostsMade.php#L40